packages/vector_graphics_compiler/lib/vector_graphics_compiler.dart

```diff
@@ -26,9 +30,79 @@ Future<VectorInstructions> parse(
   return const PaintDeduplicator().optimize(await parser.parse());
 }
 
+const VectorGraphicsCodec _codec = VectorGraphicsCodec();
+
 void main(List<String> args) async {
-  final String xml = File(args.first).readAsStringSync();
+  if (args.length != 2) {
+    print('Usage: dart vector_graphics.dart input.svg output.bin');
+    exit(1);
+  }
+  final String xml = File(args[0]).readAsStringSync();
+  final File outputFile = File(args[1]);
   final VectorInstructions instructions = await parse(xml, key: args.first);
-  // TODO: serialize instructions
-  print(instructions.commands.length);
+  final VectorGraphicsBuffer buffer = VectorGraphicsBuffer();
```

Comment:
consider moving everything below this to its own top level function like `encode` or something.